suncourse.c

```diff
@@ -0,0 +1,171 @@
+#include "dark_mode_switcher.h"
+
+#include <math.h>
+
+/*
+Based on SUNRISET.C
+Released to the public domain by Paul Schlyter, December 1992
+Source: https://stjarnhimlen.se/comp/sunriset.c
+*/
```

Comment:
I removed all comments from the original code.  
- almost every line was commented which was rather confusing than helpful  
- most of the explanations are not understandable for peaople who are not experienced with this topic, so I'd rather consider this as an implementation detail and take it as given algorithms  
- all the comments don't help much as long as the code is full of magic numbers  

However, I compared the results with several online calculators to make sure it works correctly.